include/vcpkg/base/downloads.h

```diff
@@ -20,47 +23,36 @@ namespace vcpkg::Downloads
     }
 
     void verify_downloaded_file_hash(const Files::Filesystem& fs,
-                                     const std::string& url,
+                                     const std::string& sanitized_url,
                                      const fs::path& path,
                                      const std::string& sha512);
 
-    // Returns url that was successfully downloaded from
-    std::string download_file(Files::Filesystem& fs,
-                              View<std::string> urls,
-                              View<std::string> headers,
-                              const fs::path& download_path,
-                              const std::string& sha512);
-
-    void download_file(Files::Filesystem& fs,
-                       const std::string& url,
-                       View<std::string> headers,
-                       const fs::path& download_path,
-                       const std::string& sha512);
-
     View<std::string> azure_blob_headers();
 
     std::vector<int> download_files(Files::Filesystem& fs, View<std::pair<std::string, fs::path>> url_pairs);
     ExpectedS<int> put_file(const Files::Filesystem&, StringView url, View<std::string> headers, const fs::path& file);
     std::vector<int> url_heads(View<std::string> urls, View<std::string> headers);
+    std::string replace_secrets(std::string input, View<std::string> secrets);
 
     struct DownloadManagerConfig
     {
         Optional<std::string> m_read_url_template;
         std::vector<std::string> m_read_headers;
         Optional<std::string> m_write_url_template;
         std::vector<std::string> m_write_headers;
+        std::vector<std::string> m_secrets;
         bool m_block_origin = false;
     };
 
     // Handles downloading and uploading to a content addressable mirror
-    struct DownloadManager : private DownloadManagerConfig
+    class DownloadManager
```

Comment:
We don't generally use classes in this codebase; we follow the Casey "use `struct` for a structural type, `class` for template parameters, and `typename` for disambiguation". The only exceptions I could find are either from four years ago, or modified from four years ago.

Reply:
See #102

Reply:
Sounds good